### DISCUSSION TITLE
CUDA error when I add a refiner to my (Faceswap+Lora)-Prompt, RTX4XXX, 8GB VRAM

### DISCUSSION THREAD

**Discussion by miraculix95** [lllyasviel/Fooocus#2651] [Q&A]
Hi, I am struggling with the following issue 

torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 20.00 MiB. GPU 0 has a total capacty of 4.00 GiB of which 2.92 GiB is free. Of the allocated memory 277.80 MiB is allocated by PyTorch, and 20.20 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
Total time: 132.55 seconds

RTX4XXX, on a cloud linux VM, 8GB VRAM. Ubuntu 22.04

My situation.

-   Swap space installed 64gb
-   Checking for Resizeablebar (nvidia-smi q | grep -i bar -A 3) it sows it is only 256MB, so it does not work, but activating it on Linux (using a GPU server), do not know how to do it - also maybe difficult since not my own PC (VM at a cloud provider)
-   Cuda properly installed. Driver Version: 535.129.03 CUDA Version: 12.2

The error takes place in the following situation: faceswap with imageprompt, lora used, refiner used
If I do not use a refiner it seems to be ok.

Can anyone help?

**Comment by mashb1t** [comment]
Make sure all of the GPUs VRAM (8GB) is detected (currently 4GB only, may be limited by the cloud provider though).

**Comment by Rogala** [comment]
[How to Get the Total Amount of GPU Memory](https://saturncloud.io/blog/how-to-get-the-total-amount-of-gpu-memory/)

[You have driver 535.129.03, but if you read here, since version 536.40 this has been solved by connecting shared memory.](https://nvidia.custhelp.com/app/answers/detail/a_id/5490/~/system-memory-fallback-for-stable-diffusion)
